Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -457,7 +460,7 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable, IBufferPr
             }
             set {
                 CheckBuffer();
-                if (_buffer.ReadOnly) {
+                if (@readonly) {
```

Comment:
Accessing the `readonly` property already does `CheckBuffer`, so here (and in a few cases below) the buffer is unnecessarily checked twice. I can either remove the call to `CheckBuffer` here (more consise) or replace the property access with expression `_readonlyView || _buffer.ReadOnly` (more flexibility in ordering the exception conditions, although it seems good enough now).

@slozier, any preference?

Reply:
I would prefer to keep the `CheckBuffer` since it reduces the chance of the check being accidentally removed. Is there a reason why we couldn't do something like `_isReadOnly = readonlyView || _buffer.ReadOnly` in the constructor and drop the `_readonlyView` field? That way we could just do `if (_isReadOnly)` throughout.

Reply:
It assumes that a buffer that at one time reported readonly will never become writable. I don't know if it is a valid assumption, but if you can confirm it is so, then yes, `_isReadOnly` would be the best option.

Reply:
I think it's a reasonable assumption. We're the ones defining how the buffer objects should behave, if someone chooses to disregard the contract then too bad.

In the (CPython) buffer protocol you just ask an object for a buffer to its underlying memory. If you ask for a writable buffer then:
> the exporter MUST provide a writable buffer or else report failure. Otherwise, the exporter MAY provide either a read-only or writable buffer, but the choice MUST be consistent for all consumers.

https://docs.python.org/3/c-api/buffer.html#buffer-request-types